tmt/steps/__init__.py

```diff
@@ -42,14 +42,16 @@ def __init__(self, data=None, plan=None, name=None):
             self.data = [{'name': tmt.utils.DEFAULT_NAME}]
         # Convert to list if only a single config provided
         elif isinstance(self.data, dict):
-            # Give it a name unless defined
-            if not self.data.get('name'):
-                self.data['name'] = tmt.utils.DEFAULT_NAME
             self.data = [self.data]
         # Shout about invalid configuration
         elif not isinstance(self.data, list):
             raise GeneralError(f"Invalid '{self}' config in '{self.plan}'.")
 
+        # Assign default names unless specified
+        for data in self.data:
+            if 'name' not in data:
+                data['name'] = tmt.utils.DEFAULT_NAME
```

Comment:
I'm not sure about this change. This means that mutliple prepare instances can have the same name. The name is meant to be a unique identifier of the individual steps instance/config and is used for the workdir as well. For example the following `discover` step config:

```yaml
discover:
  - how: fmf
    url: https://github.com/psss/fmf
  - how: fmf
    url: https://github.com/psss/tmt
```

would fail because tests are fetched into the same directory:

```
fatal: destination path '/var/tmp/tmt/run-094/plans/example/discover/default/tests' already exists and is not an empty directory.
```

Oher steps do not use much workdir (yet), but I guess in the future they could and we should ensure there are no collisions.